src/main/kotlin/org/jetbrains/research/transformations/Anonymization.kt

```diff
@@ -0,0 +1,75 @@
+@file:Suppress("UNCHECKED_CAST")
+
+package org.jetbrains.research.transformations
+
+import com.github.gumtreediff.tree.ITree
+import com.github.gumtreediff.tree.Tree
+import com.github.gumtreediff.tree.TreeContext
+
+object Anonymization: Transformation {
+    override val metadataKey = "anonymization"
+    private var anonymNamesMap: MutableMap<String, String> = mutableMapOf()
+    private var numOfVariables = 0
+    private const val VAR_PREFIX = "v"
+
+
+    override fun apply(treeCtx: TreeContext, toStoreMetadata: Boolean) {
+        anonymNamesMap = mutableMapOf()
+        numOfVariables = 0
+        val tree = treeCtx.root
+        val treeIterator = tree.trees.iterator()
+
+        while (treeIterator.hasNext()) {
+            val node = treeIterator.next()
+
+            if (treeCtx.getTypeLabel(node) == "NameStore") {
+                anonymNamesMap.getOrPut(node.label, { setAnonLabel(node, toStoreMetadata) })
+            }
+
+            if (treeCtx.getTypeLabel(node) == "NameLoad") {
+                if (node.label in anonymNamesMap.keys) {
+                    val newLabel = anonymNamesMap[node.label]
+                    if (toStoreMetadata) node.setMetadata(metadataKey, mutableMapOf(newLabel to node.label))
+                    node.label = newLabel
+                }
+
+            }
+        }
+
+        return
```

Comment:
Why do you use it? I think you can delete it

Reply:
done :)